shared/settings/notifications/index.js

```diff
@@ -1,41 +1,65 @@
 // @flow
 import React from 'react'
-import {Box, Button, Checkbox, HeaderHoc, ProgressIndicator, Text} from '../../common-adapters'
-import {isMobile} from '../../constants/platform'
+import {Box, Button, Checkbox, ProgressIndicator, Text} from '../../common-adapters'
 import {globalStyles, globalMargins} from '../../styles'
 
 import type {Props} from './index'
 
+const makeCheckbox = (
+  group: string,
+  s: {name: string, description: string, subscribed: boolean},
+  props: Props
+) => (
+  <Checkbox
+    style={{marginRight: globalMargins.medium, marginTop: globalMargins.tiny}}
+    key={s.name}
+    disabled={!props.allowEdit}
+    onCheck={() => props.onToggle(group, s.name)}
+    checked={s.subscribed}
+    label={s.description}
+  />
+)
+
+const renderGroup = (groupName: string, props: Props) => {
+  const labels = {
```

Comment:
I think this stuff should be passed in, and we dont' need to pass props itself in, just what this thing needs. I'd imagine the props this component gets are like
```
<Group
title='Email me:',
unsub='mail' // (so we put the 'Unsubscribe me from ${unsub}')
settings={this.props.groups.app_push.settings}
allowEdit={this.props.allowEdit}
onToggle={settings => this.props. onToggleUnsubscribe(settings) ... etc}
/>

```